src/sortStudents.ts

```diff
@@ -1,16 +1,64 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[]
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverage(numbers: number[]): number {
+  return numbers.reduce((sum: number, num: number) => sum + num, 0)
+  / numbers.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((a: Student, b: Student): number => {
+    const firstAverage = getAverage(a[SortType.AverageGrade]);
+    const secondAverage = getAverage(b[SortType.AverageGrade]);
+
+    switch (sortBy) {
+      case SortType.Married:
+        return order === 'asc'
+          ? Number(a[SortType.Married]) - Number(b[SortType.Married])
+          : Number(b[SortType.Married]) - Number(a[SortType.Married]);
+
+      case SortType.Name:
+        return order === 'asc'
+          ? a[SortType.Name].localeCompare(b[SortType.Name])
+          : b[SortType.Name].localeCompare(a[SortType.Name]);
+
+      case SortType.Surname:
+        return order === 'asc'
+          ? a[SortType.Surname].localeCompare(b[SortType.Surname])
+          : b[SortType.Surname].localeCompare(a[SortType.Surname]);
```

Comment:
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return order === 'asc'
          ? a[sortBy].localeCompare(b[sortBy])
          : b[sortBy].localeCompare(a[sortBy]);
```
Combine the similar cases
![image](https://user-images.githubusercontent.com/96409563/229454172-e8a4343c-4aeb-4d78-9320-690d646d602c.png)